src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private int initialCapacity = 16;
+    private float loadFactor = 0.75f;
+    private int size = 0;
+    private LinkedList<Entry<K, V>>[] buckets;
+    private Entry<K, V> nullKeyEntry;
+
+    public MyHashMap() {
+        buckets = new LinkedList[initialCapacity];
+    }
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
 
-    @Override
     public void put(K key, V value) {
+        if (key == null) {
+            if (nullKeyEntry == null) {
+                size++;
+            }
+            nullKeyEntry = new Entry<>(key, value);
+            return;
+        }
+
+        int index = getBucketIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new LinkedList<>();
+        }
+
+        for (Entry<K, V> entry : buckets[index]) {
+            if (entry.key.equals(key)) {
+                entry.value = value;
+                return;
+            }
+        }
 
+        buckets[index].add(new Entry<>(key, value));
+        size++;
+
+        if (size >= initialCapacity * loadFactor) {
+            resize();
+        }
     }
 
-    @Override
     public V getValue(K key) {
```

Comment:
The method `getValue` is public and should be before any private methods or inner classes according to the checklist.